assets/pages/oneoff-contributions/helpers/ajax.js

```diff
@@ -92,19 +92,21 @@ function postToEndpoint(request: Object, dispatch: Function, abParticipations: P
       dispatch(checkoutSuccess());
     }
     return response.json();
-  }).then((responseJson) => {
-    if (responseJson.error.exceptionType === 'CardException') {
-      dispatch(checkoutError('Your card has been declined.'));
-    } else {
-      const errorHttpCode = responseJson.error.errorCode || 'unknown';
-      const exceptionType = responseJson.error.exceptionType || 'unknown';
-      const errorName = responseJson.error.errorName || 'unknown';
-      logException(`Stripe payment attempt failed with following error: code: ${errorHttpCode} type: ${exceptionType} error-name: ${errorName}.`);
-      dispatch(checkoutError('There was an error processing your payment. Please\u00a0try\u00a0again\u00a0later.'));
+  }).then((data) => {
+    if (!(data.type === 'success')) {
+      if (data.type === 'error') {
+        const { error: { exceptionType = 'Unknown', responseCode = 0, errorName = 'Unknown' } } = data;
```

Comment:
There is a lot of nesting here which makes it hard to understand, I would extract everything inside the `if (data.type === 'error')` block into a separate function, something like `handleError`